### PR TITLE
Reset state during broken period, remove breaking changelog loop early

### PR DESCRIPTION
**Pull request by aaboyd**
# Description of change
Reset state for time period where it was broken.

Drop "break" during changelog ingestion to get them all

# Manual QA steps
 - [x] Modified date for the botched state timeframe, verified issues were pulled 
 - [x] Ran with hardcoded issue key to verify changelogs were in the output
 - [x] Ran in production for minware to verify data was picked up and updated appropriately

# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_jira/streams.py [hunk 1]
```diff
@@ -77,9 +77,6 @@ def sync_sub_streams(page, issue_changelog_updated):
                 # when expanding changelogs for an issue, jira returns 100
                 if changelog_response['maxResults'] >= changelog_response['total']:
                     for changelog in changelogs:
-                        if utils.strptime_to_utc(changelog["created"]) > issue_changelog_updated:
```
Comment (aaboyd) [inline tap_jira/streams.py:80]: Removing this completely, which reverts back to previous behaviour.   The condition is inverted here.  Previously was focused on fetching more changelogs and the other branch of code was running in my test issues.

File: tap_jira/streams.py [hunk 2]
```diff
@@ -376,6 +373,9 @@ def sync_project(self, fieldNames, knownFields, project_key_or_id = None):
         last_updated = Context.update_start_date_bookmark(updated_bookmark)
         timezone = Context.retrieve_timezone()
         start_date = last_updated.astimezone(pytz.timezone(timezone)).strftime("%Y-%m-%d %H:%M")
+        if datetime.datetime(2024, 5, 17, 0, 0, tzinfo=pytz.utc) < last_updated < datetime.datetime(2024, 7, 23, 5, 0, 0, tzinfo=pytz.utc):
```
Comment (aaboyd) [inline tap_jira/streams.py]: These ranges are pretty liberal, want to ensure that we don't miss any data.  Its possible that we will reingest ~5 days of data once or twice depending on when orgs run.

File: tap_jira/streams.py (hunk 1 above)
Comment (aaboyd) [reply tap_jira/streams.py:80]: This takes out the need to repair the issue_changelog state.  The condition is correct in the branch that fetches additional changelogs.  Its only broken when all changelogs are included

**Review by KBorders01 — APPROVED** [review]

File: tap_jira/streams.py [hunk 3]
```diff
@@ -68,39 +68,32 @@ def sync_sub_streams(page, issue_changelog_updated):
         if Context.is_selected(CHANGELOGS.tap_stream_id):
             changelog_response = issue.pop("changelog")
             changelogs = changelog_response["histories"]
-            if any(
```
Comment (aaboyd) [inline tap_jira/streams.py:71]: Needed to also remove this any check to get older changelogs

**Pull request merged**